scripts/devenv-builder/create-vm.sh

```diff
@@ -52,6 +49,12 @@ if [ "${SWAPSIZE}" -eq 0 ] ; then
     sed -i "s;^part swap;#part swap;" "${KICKSTART_FILE}"
 fi
 
+# Allow the VM creation to run in parallel using multiple instances of this script
+# Note: If 'dnf' command is run in parallel, its database is corrupted
+# === Start critical section ===
+exec {LOCK_FD}<"$0"
+flock --exclusive ${LOCK_FD}
+
 sudo dnf install -y libvirt virt-manager virt-install virt-viewer libvirt-client qemu-kvm qemu-img sshpass
```

Comment:
Could we move the dnf commands to another script that is only run once? We have a couple of different configure scripts that could be candidates, or we could make a new one.